contracts/Controller.sol

```diff
@@ -339,7 +356,22 @@ contract Controller is Initializable, OwnableUpgradeSafe, ReentrancyGuardUpgrade
      */
     function operate(Actions.ActionArgs[] memory _actions) external nonReentrant notFullyPaused {
         (bool vaultUpdated, address vaultOwner, uint256 vaultId) = _runActions(_actions);
-        if (vaultUpdated) _verifyFinalState(vaultOwner, vaultId);
+        if (vaultUpdated) {
+            _verifyFinalState(vaultOwner, vaultId);
+            vaultLatestUpdate[vaultOwner][vaultId] = now;
+        }
+    }
+
+    /**
+     * @notice sync vault latest update timestamp
+     * @dev anyone can update the latest time the vault was touched by calling this function
+     * vaultLatestUpdate will sync if the vault is well collateralized
+     * @param _owner vault owner address
+     * @param _vaultId vault id
+     */
+    function sync(address _owner, uint256 _vaultId) external nonReentrant notFullyPaused {
```

Comment:
Is it useful to have a canSync public function or just do via f/e calls or require devs monitoring via bots to do it themselves?

Reply:
what `canSync()` for ? to check if it is able to sync? if so it will be basically the same as `sync()` that will check the margin and return bool. We can keep it in the frontend as it can be done by checking the margin state.

Reply:
@aleone, it looks like the equivalent of `canSync` would be calling `calculator.getExcessCollateral`:

```solidity
(, bool isValidVault) = calculator.getExcessCollateral(vault, typeVault);
```

I think a separate `canSync` function could be useful later on, but don't think it's necessary right now (for example, a FE could poll the vault state for a UI).